@xen-orchestra/xapi/vdi.mjs

```diff
@@ -76,6 +76,11 @@ class Vdi {
       })
     })
 
+    if (gm?.PV_drivers_detected) {
+      // It may happen that `PV_drivers_detected` returns `true` too quickly, and thus we destroy the cloudConfig VDI before the first boot
```

Comment:
The issue lies in the boot process order: guest tools may load before CloudInit, causing the drive to be removed prematurely, before CloudInit reads the configuration. Since we can't easily modify CloudInit to signal completion, we must allow enough time for the OS to boot and for CloudInit to process the config before removing the drive.